var/spack/repos/builtin/packages/bohrium/package.py

```diff
@@ -220,3 +223,67 @@ def setup_environment(self, spack_env, run_env):
         # the self.prefix.include dir
         run_env.prepend_path("CPATH", self.prefix.include.bohrium)
         run_env.set("BH_CONFIG", self.config_file)
+
+    #
+    # Quick tests
+    #
+    @run_after('install')
+    @on_package_attributes(run_tests=True)
+    def check_install(self):
+        spec = self.spec
+        test_env = {}
+
+        # Make sure the correct config is found
+        test_env["BH_CONFIG"] = self.config_file
+
+        # Remove the lib/spackenv directory from the PATH variable when
+        # executing the tests, becauses it messes with the JIT compilation
+        # inside Bohrium
+        paths = os.environ['PATH'].split(':')
+        paths = [p for p in paths if "spack/env" not in p]
+        test_env["PATH"] = ":".join(paths)
+
+        # Add the PYTHONPATH to bohrium to the PYTHONPATH environment
+        pythonpaths = [p for p in os.environ["PYTHONPATH"].split(":")]
+        pythonpaths.append(join_path(self.prefix,
+                                     spec['python'].package.site_packages_dir))
+        test_env["PYTHONPATH"] = ":".join(pythonpaths)
+
+        # Collect the stacks which should be available:
+        stacks = ["default"]
+        if "+openmp" in spec:
+            stacks.append("openmp")
+        if "+cuda" in spec:
+            stacks.append("cuda")
+        if "+opencl" in spec:
+            stacks.append("opencl")
+
+        # C++ compiler and compiler flags
+        cpp = which("c++")
```

Comment:
Use `spec.compiler.cxx`!

Reply:
Good point. I somewhat blindly copied code from the `compile_c_and_execute` function of  `package_test.py`. On that note: Do you think it would be sensible to provide such a function for c++ as well?